CRM/Contact/BAO/Query.php

```diff
@@ -2867,7 +2867,7 @@ public function includeContactSubTypes($value, $grouping, $op = 'LIKE') {
     }
     elseif (is_array($value)) {
       foreach ($value as $k => $v) {
-        if (!empty($k)) {
+        if (isset($k)) {
```

Comment:
hey there, i had a look at this last night. it looks like the reason for the original (!empty($k)) is lost in the sands of time/svn. It would sometimes evaluate to false, but not sure why. Since (isset($k)) will always evaluate to true, agree that it makes sense to not add it at all.

I thought adding a test might allow us to stop worrying about why it was added in the first place.